src/components/videoCard/VideoCard.jsx

```diff
@@ -35,8 +60,36 @@ const VideoCard = (props) => {
             </div>
           </div>
         </div>
-        <i className="fas fa-ellipsis-v"></i>
+        <i
+          className="fas fa-ellipsis-v"
+          onClick={() => setShowOptionsDropdown((prev) => !prev)}
+        ></i>
       </div>
+      {showOptionsDropdown && (
+        <ul className="options__dropdown">
+          {isVideoLiked ? (
+            <li onClick={likeHandler}>
+              <i className="fas fa-heart"></i>Remove from likes
+            </li>
+          ) : (
+            <li onClick={likeHandler}>
+              <i className="far fa-heart"></i>Add to likes
+            </li>
+          )}
+          {isVideoInWatchLater ? (
+            <li>
+              <i className="fas fa-clock"></i>Remove from watch Later
+            </li>
+          ) : (
+            <li>
+              <i className="far fa-clock"></i>Add to Watch Later
+            </li>
+          )}
+          <li>
+            <i className="fas fa-plus"></i>Save to playlist
+          </li>
+        </ul>
+      )}
```

Comment:
Since you are using ternary operators in the return statement, for better readability  - 

Recently I learned a way to make our code inside return look clean. What you can do is, before the return statement where we write all the logic you can make components. For example -

export default func XYZ(){

const yourComponent = (

write all your jsx here

)

return (

yourComponent

)
}

Reply:
Thanks for suggesting, that I will do abstraction at the end.